### PR TITLE
Contributions: preserving  product_type

### PR DESCRIPTION
**Pull request by dparikh**
Store and pass`product_type` in the `subscriptionResponse` for `payconfirmiframe` to use.

This allows changes to the payconfirmiframe page to render the different text/label.

### CODE REVIEW

File: src/runtime/pay-flow-test.js [hunk 1]
```diff
@@ -281,14 +285,15 @@ describes.realWin('PayCompleteFlow', {}, env => {
     jserrorMock.verify();
   });
 
-  it('should have valid flow constructed', () => {
+  it('should have valid flow constructed ******', () => {
```
Comment (chenshay) [inline src/runtime/pay-flow-test.js]: what are all these stars for?

File: src/runtime/pay-flow-test.js [hunk 2]
```diff
@@ -281,14 +285,15 @@ describes.realWin('PayCompleteFlow', {}, env => {
...
     const purchaseData = new PurchaseData();
     const userData = new UserData('ID_TOK', {
       'email': 'test@example.org',
     });
     const entitlements = new Entitlements('service1', 'RaW', [], null);
     const response = new SubscribeResponse(
-        'RaW', purchaseData, userData, entitlements);
+        'RaW', purchaseData, userData, entitlements,
+        ProductType.SUBSCRIPTION, entitlements);
```
Comment (chenshay) [inline src/runtime/pay-flow-test.js]: entitlements is here twice. You need to remove one of them

File: src/runtime/pay-flow-test.js [hunk 3]
```diff
@@ -332,6 +338,7 @@ describes.realWin('PayCompleteFlow', {}, env => {
           _client: 'SwG $internalRuntimeVersion$',
           publicationId: 'pub1',
           loginHint: 'test@example.org',
+          productType: undefined,
```
Comment (chenshay) [inline src/runtime/pay-flow-test.js]: is it supposed to be null or undefined or doesn't matter?

File: src/runtime/pay-flow-test.js (hunk 1 above)
Comment (dparikh) [reply src/runtime/pay-flow-test.js]: leftover from debugging...Removed.

File: src/runtime/pay-flow-test.js (hunk 2 above)
Comment (dparikh) [reply src/runtime/pay-flow-test.js]: Removed.

File: src/runtime/pay-flow-test.js (hunk 3 above)
Comment (dparikh) [reply src/runtime/pay-flow-test.js]: Here `productType: undefined` is correctly matched.

**Pull request merged**